### DISCUSSION TITLE
Contact modeling with mutiple surfaces

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#19304] [Q&A Modules: General] [answered]
I am running a simulation with contact modeling. I have defined motion for a single particle that may interact with one of 10 surfaces in time. I am getting the following error:

ERROR: negative Jacobian -3.29043e-24 at point (x,y,z)=(0.016665, 0.013855, 0.0204749) in element 33771
Stack frames: 37

What should be my focus to help eliminate this particular error?

**Comment by recuero** [comment, marked as answer]
The error points to an inverted element. It's not straightforward to know what best to do here without more information, this error can be caused by a multitude of issues. If you know your contact settings are good, I would try reducing the time step.

**Reply by GiudGiud** [reply]
I would also go look at that element at the beginning of the simulation and keep tracking it in the exodus output, see how its shape evolves. Maybe the starting shape is pretty skewed, maybe it looks fine then gets squished flat in contact
the Jacobian is really really small so I think it s likely to have been flattened
